### ISSUE TITLE
Feature Request: Support idrive.com as a backend

### ISSUE THREAD

**Issue by zappe** [duplicati/duplicati#3678]
Would be great if this was supported as an backend.

**Comment by Pectojin** [comment]
I can't seem to find the API documentation.

The only thing I could find is this https://idrivesync.com/evs/web-developers-guide.htm and I'm not sure it's the same company.

**Comment by zappe** [comment]
I don't think so.

Maybe this is it, https://github.com/idrivevangelist?

I've also sent them a mail.

**Comment by Pectojin** [comment]
I ran into those too. They look like the right company but I'm having a bit of trouble finding clear instructions on e.g. Auth, get, and put operations from the examples.

**Comment by zappe** [comment]
They got back to me via email and said they didn't have an API.
So I guess we can close this.

**Comment by Pectojin** [comment]
Oh, that's unfortunate. 

Thanks for taking contact to them to verify :)

**Comment by Fabeson** [comment]
I think they offer 2 different products: (regular) iDrive with their own proprietary sync clients and also **iDrive Cloud** which actually uses S3 APIs - Just putting this here for future reference :)

- https://www.idrive.com/cloud/guides/default

**Comment by dkrahmer** [comment]
@Pectojin I have created an IDrive backend that I would like to submit a PR for. Should this issue be re-opened or should I create a new one?

**Comment by dkrahmer** [comment]
Before building the backend, I contacted IDrive and asked about how to authenticate to use the IDrive Sync API. This is their exact reply: Sorry APIs is not supported by Idrive.

I was able to reverse engineered the authentication logic from code found in the IDriveForLinux PERL scripts provided by IDrive. Download from: https://www.idrive.com/linux-backup-scripts

It requires the use of some undocumented IDrive APIs to authenticate. Once the authentication steps are complete all of the fully documented IDriveSync APIs are used.
